crates/sui-framework/sources/immutable_external_resource.move

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Sui types for specifying off-chain/external resources.
+///
+/// The keywords "MUST", "MUST NOT", "SHOULD", "SHOULD NOT" and "MAY" below should be interpreted as described in
+/// RFC 2119.
+///
+module sui::immutable_external_resource {
+    use sui::digest::Sha2256Digest;
+    use sui::url::{Url, inner_url};
+
+    /// ImmutableExternalResource: An arbitrary, mutable URL plus an immutable digest of the resource.
+    ///
+    /// Represents a resource that can move but must never change. Example use cases:
+    /// - NFT images.
+    /// - NFT metadata.
+    ///
+    /// `url` MUST follow RFC-3986. Clients MUST support (at least) the following schemes: ipfs, https.
+    /// `digest` MUST be set to SHA256(content of resource at `url`).
+    ///
+    /// Clients of this type MUST fetch the resource at `url`, compute its digest and compare it against `digest`. If
+    /// the result is false, clients SHOULD indicate that to users or ignore the resource.
+    struct ImmutableExternalResource has store, copy, drop {
```

Comment:
Before we merge we need to agree on the hash function supported. 
There are at least two reasons for this:
- (agility) i.e. IPFS links include the hash algorithm via the multi-codec format, see https://proto.school/anatomy-of-a-cid/02 and https://docs.filebase.com/ipfs/ipfs-cids (note the two CID Formats. There are currently two different versions of IPFS CIDs - version 0 (0v) and version 1 (v1).
- Sui's default hash is sha3, not sha2

TL;DR I'd prefer we take a decision on these matters in the next crypto meeting and then explain in the docs or code comments why this X or Y re supported functions.